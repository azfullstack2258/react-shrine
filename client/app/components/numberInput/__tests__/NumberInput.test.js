/* global describe, beforeAll, it, expect */
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { createShallow } from '@material-ui/core/test-utils';
import Wrapper, { NumberInput } from '../index';

configure({ adapter: new Adapter() });

describe('NumberInput', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow({ untilSelector: NumberInput });
  });

  it('renders correctly', () => {
    const initial = 1;
    const min = 1;
    const max = 5;
    const wrapper = shallow(<Wrapper initial={initial} min={min} max={max} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
