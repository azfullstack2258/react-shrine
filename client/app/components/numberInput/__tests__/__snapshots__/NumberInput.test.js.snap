// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders correctly 1`] = `
<div
  className="NumberInput-container-1"
>
  <WithStyles(Button)
    classes={
      Object {
        "outlined": "NumberInput-leftButton-3",
        "root": "NumberInput-button-2",
      }
    }
    onClick={[Function]}
    variant="outlined"
  >
    <pure(RemoveIcon) />
  </WithStyles(Button)>
  <WithStyles(Input)
    classes={
      Object {
        "input": "NumberInput-input-5",
        "root": "NumberInput-inputRoot-6",
      }
    }
    disableUnderline={true}
    value={1}
  />
  <WithStyles(Button)
    classes={
      Object {
        "outlined": "NumberInput-rightButton-4",
        "root": "NumberInput-button-2",
      }
    }
    onClick={[Function]}
    variant="outlined"
  >
    <pure(AddIcon) />
  </WithStyles(Button)>
</div>
`;
